awx_collection/plugins/module_utils/tower_api.py

```diff
@@ -551,12 +635,12 @@ def create_if_needed(self, existing_item, new_item, endpoint, on_create=None, it
                 self.modify_associations(sub_endpoint, associations[association_type])
```

Comment:
You're adding a helper method to tease out "detail" or "__all__" or other stuff from the response, and to the extent that you still want to do that, branching like what we have here in `create_if_needed` should be converted to use that helper method.

Reply:
Also, the method `create_if_needed` calls `modify_associations` while the other one calls `process_associations`. If it's going to be called in exactly 1 place, there's no reason to put it into a separate method.